gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/computer/VertexProgram.java

```diff
@@ -48,6 +50,10 @@
 
     public static final String VERTEX_PROGRAM = "gremlin.vertexProgram";
 
+    default List<Pair<String, Class>> getVertexPropertyKeys() {
```

Comment:
Is there a reason you can't get this information from `getVertexComputeKeys()`? Seems like it would be better if `VertexComputeKey` was modified to include the data type which is what you seem to want to add.  It feels like, with this change you'd be making the `VertexProgram` developer define the keys twice which leaves open the possibility for divergence.

Here's an example from this PR in `ConnectedComponentVertexProgram`:

```java
@Override
    public List<Pair<String, Class>> getVertexPropertyKeys() {
        return Arrays.asList(
            org.apache.commons.lang3.tuple.Pair.of(HALTED_TRAVERSERS, Object.class),
            org.apache.commons.lang3.tuple.Pair.of(ACTIVE_TRAVERSERS, Object.class),
            org.apache.commons.lang3.tuple.Pair.of(COMPONENT, Object.class));
    }
```

It defines `COMPONENT` when i think it should be `property`.

I think using the existing `getVertexComputeKeys()` approach also would solve the testing issue that @mikepersonick brought up because that method is implicitly tested already - you wouldn't really need to add anything new. Perhaps you would add some basic unit test to validate the new behavior in the `VertexComputeKey`?

You targeted `3.5-dev` so you need to do this in a fashion that is non-breaking. I sense that is possible if you add a new `VertexComputeKey.of(String, boolean, Class<?>)` and have the old `of(String,boolean)` call the new method with `Object.class` for the last argument. Then you could go through all of the existing `VertexProgram` instances and set the types explicitly. I  don't think that will break anything. 

Assuming this all makes sense and I'm not missing anything, please make sure to include a CHANGELOG entry when you're done.

Reply:
> It defines COMPONENT when i think it should be property

You are right, I was wrong. I used the default property name in the test so didn't catch this bug.

> Is there a reason you can't get this information from getVertexComputeKeys()

Thanks, I totally forgot about it 😅 . That should mostly suffice the use case I described, except that I am not 100% sure why other vertex programs don't contain `HALTED_TRAVERSERS` and `ACTIVE_TRAVERSERS` in `vertexComputeKeys`. Anyways, this PR is not very useful now so I am closing it. Thank you @mikepersonick and @spmallette for your reviews!